spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepositoryTests.java

```diff
@@ -353,8 +352,9 @@ public void findAllSortAsc() {
 	}
 
 	@Test
-	public void deleteAllByIdUnimplemented() {
-		assertThatThrownBy(() -> this.simpleDatastoreRepository.deleteAllById(new ArrayList<>()))
-				.isInstanceOf(UnsupportedOperationException.class);
+	public void deleteAllById() {
+		List<String> keys = Arrays.asList("1", "2");
+		this.simpleDatastoreRepository.deleteAllById(keys);
+		verify(this.datastoreTemplate).deleteAllById(keys, Object.class);
```

Comment:
This unit test only verifies `deleteAllById` of datastoreTemplate mock is called.
Test for the actual methods is already tested in [DatastoreTemplateTests](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplateTests.java#L923-L938)

Reply:
That's good, even if it feels weird. It's a unit test that only tests the functionality that you changed in this class.
You've added an integration test to validate end-to-end functionality, so it's completely fair to only have the unit test do this much.